types/node/stream.d.ts

```diff
@@ -57,6 +57,11 @@ declare module 'stream' {
              * @since v16.8.0
              */
             static isDisturbed(stream: Readable | NodeJS.ReadableStream): boolean;
+            /**
+             * Returns whether the stream is readable.
+             * @since v17.4.0
+             */
+            static isReadable(stream: unknown): stream is Readable;
```

Comment:
according to the docs: https://nodejs.org/api/stream.html#streamisreadablestream some type of `stream` is expected as parameter.

although passing anything but a stream returns `null`

```ts
require('stream').isReadable('foo') // null
```

interestingly, only passing a `ReadableStream` returns `true`, passing a `WriteableStream` returns also `null`, which might be a bug.

Reply:
Looks like I've misunderstood the intended functionality. Looking at the code in the [original PR](https://github.com/nodejs/node/pull/41199/files), it's designed to return whether a readable stream is in a readable state, not whether the stream is a readable stream.

Oh well, this makes the function not so useful for my intended use case (even though by chance it works anyway 😄 ).

I'll try to update the PR in the coming days.

Reply:
I'm thinking the function should probably throw an error if anything else but a ReadableStream is passed in.